### PR TITLE
First PR to fix some bugs

### LINKED ISSUE #1657: Analyzer manager failed to retrieve bin file to map()
Please see this PR and also copy/paste issue below
https://github.com/InstituteforDiseaseModeling/malaria-mmc-diagnostics-idmtools/pull/11

prevalence_analyzer.py fail with error:

C:\mnc_idmtools_37\Scripts\python.exe C:/github/malaria-mmc-diagnostics-idmtools/simulation/analyzers/run_prevalence_analyzer.py
/!\ WARNING: File 'idmtools.ini' Not Found! For details on how to configure idmtools, see https://docs.idmod.org/projects/idmtools/en/v1.6.4/configuration.html for details on how to configure idmtools.

[BELEGOST]
{
   "endpoint": "https://comps.idmod.org",
   "environment": "Belegost"
}
  0%|          | 0/5000 [00:00<?, ?it/s]Analyze Manager
 | 5000 item(s) selected for analysis
 | partial_analyze_ok is False, max_items is None, and 0 item(s) are being ignored
 | Analyzer(s): 
 |  - PrevalenceAnalyzer File parsing: on / Use cache: off
 | Pool of 8 analyzing process(es)
100%|█████████▉| 4981/5000 [00:12<00:00, 632.28it/s]C:\mnc_idmtools_37\lib\site-packages\idmtools\entities\iplatform.py:121: UserWarning: Please use Factory to create Platform! For example: 
    platform = Platform('COMPS', **kwargs)
  warnings.warn("Please use Factory to create Platform! For example: \n    platform = Platform('COMPS', **kwargs)")
100%|██████████| 5000/5000 [00:14<00:00, 344.40it/s]
concurrent.futures.process._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "c:\python37\lib\concurrent\futures\process.py", line 239, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
  File "C:\mnc_idmtools_37\lib\site-packages\idmtools\analysis\map_worker_entry.py", line 41, in map_item
    return _get_mapped_data_for_item(item, analyzers, platform)
  File "C:\mnc_idmtools_37\lib\site-packages\idmtools\analysis\map_worker_entry.py", line 101, in _get_mapped_data_for_item
    raise e
  File "C:\mnc_idmtools_37\lib\site-packages\idmtools\analysis\map_worker_entry.py", line 92, in _get_mapped_data_for_item
    selected_data[analyzer.uid] = analyzer.map(data, item)
  File "C:\github\malaria-mmc-diagnostics-idmtools\simulation\analyzers\prevalence_analyzer.py", line 49, in map
    data[f'output/SpatialReportMalariaFiltered_{self.spatial_channels[0]}.bin'], self.spatial_channels[0])
  File "C:\github\malaria-mmc-diagnostics-idmtools\simulation\analyzers\prevalence_analyzer.py", line 8, in construct_spatial_output_df
    n_nodes = rawdata['n_nodes']
TypeError: 'NoneType' object is not subscriptable
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:/github/malaria-mmc-diagnostics-idmtools/simulation/analyzers/run_prevalence_analyzer.py", line 30, in <module>
    manager.analyze()
  File "C:\mnc_idmtools_37\lib\site-packages\idmtools\analysis\analyze_manager.py", line 457, in analyze
    map_results, status = self._run_and_wait_for_mapping(executor)
  File "C:\mnc_idmtools_37\lib\site-packages\idmtools\analysis\analyze_manager.py", line 338, in _run_and_wait_for_mapping
    raise ex
TypeError: 'NoneType' object is not subscriptable
'NoneType' object is not subscriptable
Process finished with exit code 1
This is due to a bug in idmtools which can not load bin file to map's data (still reference dtk)
https://github.com/InstituteforDiseaseModeling/idmtools/blob/240f8095a494fa4f2cccf235fb82b92d8e47f195/idmtools_core/idmtools/analysis/map_worker_entry.py#L85
https://github.com/InstituteforDiseaseModeling/idmtools/blob/240f8095a494fa4f2cccf235fb82b92d8e47f195/idmtools_core/idmtools/utils/file_parser.py#L51
https://github.com/InstituteforDiseaseModeling/idmtools/blob/240f8095a494fa4f2cccf235fb82b92d8e47f195/idmtools_core/idmtools/utils/file_parser.py#L146

### PR DESCRIPTION
**Pull request by ZDu-IDM**
Fixed
#1657
#1925
#2138
#2137
#2135
#2136
#2140

### CODE REVIEW

File: idmtools_platform_comps/idmtools_platform_comps/comps_operations/experiment_operations.py [hunk 1]
```diff
@@ -262,7 +262,7 @@ def post_create(self, experiment: Experiment, **kwargs) -> NoReturn:
             user_logger.log(SUCCESS, f"\nThe created experiment can be viewed at {self.platform.endpoint}/#explore/"
                                      f"Simulations?filters=ExperimentId={experiment.uid}\nSimulations are still being created\n"
                             )
-        experiment.post_creation(self.platform)
+        super().post_creation(self.platform, **kwargs)
```
Comment (shchen-idmod) [inline idmtools_platform_comps/idmtools_platform_comps/comps_operations/experiment_operations.py:265]: there is no post_creation in super class.

**Review by shchen-idmod — CHANGES_REQUESTED** [review]

File: idmtools_core/idmtools/analysis/platform_anaylsis.py [hunk 2]
```diff
@@ -45,6 +45,7 @@ def __init__(self, platform: IPlatform, analyzers: List[Type[IAnalyzer]],
         Args:
             platform: Platform
             experiment_ids: Experiment ids
+            simulaiton_ids: Simulation ids
```
Comment (shchen-idmod) [inline idmtools_core/idmtools/analysis/platform_anaylsis.py:48]: typo: simulaiton_ids, should be simulation_ids

File: idmtools_platform_comps/idmtools_platform_comps/comps_operations/experiment_operations.py (hunk 1 above)
Comment (ZDu-IDM) [reply idmtools_platform_comps/idmtools_platform_comps/comps_operations/experiment_operations.py]: Fixed, good catch!

File: idmtools_core/idmtools/analysis/platform_anaylsis.py (hunk 2 above)
Comment (ZDu-IDM) [reply idmtools_core/idmtools/analysis/platform_anaylsis.py]: Fixed, good catch!

**Comment by shchen-idmod** [comment]
This new test failed: You can add to comps test file to the end: test_python_experiment.py

 ```
def test_experiment_hooks(self):
        base_task = CommandTask(command="python --version")
        sim = Simulation(task=base_task)

        exp = Experiment(name='ExpHooks')
        exp.simulations = [sim]

        def add_pre_exp_tag(item: Experiment, platform: 'COMPSPlatform'):
            item.tags['pre_tag'] = "pre"

        def update_post_exp_tags(item: Experiment, platform: 'COMPSPlatform'):
            tags = {"post_tag": "post"}
            update_item(platform, exp.id, ItemType.EXPERIMENT, tags)

        exp.add_pre_creation_hook(add_pre_exp_tag)
        exp.add_post_creation_hook(update_post_exp_tags)

        exp.run(wait_until_done=True)
        tag_value = "idmtools.entities.command_task.CommandTask"
        exp_tags = {'pre_tag': "pre", 'post_tag': 'post', 'task_type': tag_value}
        self.validate_exp_tags(exp.id, exp_tags)

    def validate_exp_tags(self, exp_id, expected_tags):
        tags = COMPSExperiment.get(exp_id, QueryCriteria().select_children('tags')).tags
        tags.pop('idmtools')
        self.assertDictEqual(tags, expected_tags)
```

File: idmtools_platform_comps/idmtools_platform_comps/comps_operations/experiment_operations.py [hunk 3]
```diff
@@ -262,7 +262,7 @@ def post_create(self, experiment: Experiment, **kwargs) -> NoReturn:
             user_logger.log(SUCCESS, f"\nThe created experiment can be viewed at {self.platform.endpoint}/#explore/"
                                      f"Simulations?filters=ExperimentId={experiment.uid}\nSimulations are still being created\n"
                             )
-        experiment.post_creation(self.platform)
+        super().post_create(self.platform, **kwargs)
```
Comment (shchen-idmod) [inline idmtools_platform_comps/idmtools_platform_comps/comps_operations/experiment_operations.py:265]: Actually previous line was good: experiment.post_creation(self.platform)

**Comment by shchen-idmod** [comment]
For log bug https://github.com/InstituteforDiseaseModeling/idmtools/issues/2137
with test:
```
def test_console_off_log_off1(self):
        ini_content = f"""
        [Logging]
            level = CRITICAL
            console = on
            enable_file_logging = off
            user_output = on
        """
        self.create_temp_ini(ini_content)
        IdmConfigParser(file_name=self.tempfile_ini.name)
        self.platform = Platform('SLURMSTAGE')
        logger_demo.write_some_logs(user=True, root=True, comps=True, exp=True)
```
Used to print: 
2023-12-04 17:09:26,172 BMGF-R913QDCE root[57444] CRITICAL root: 5
2023-12-04 17:09:26,173 BMGF-R913QDCE user[57444] CRITICAL user: 55

Now print:
2023-12-04 17:10:06,146 BMGF-R913QDCE root[11904] DEBUG root: 1
2023-12-04 17:10:06,146 BMGF-R913QDCE root[11904] INFO root: 2
2023-12-04 17:10:06,146 BMGF-R913QDCE root[11904] WARNING root: 3
2023-12-04 17:10:06,147 BMGF-R913QDCE root[11904] ERROR root: 4
2023-12-04 17:10:06,147 BMGF-R913QDCE root[11904] CRITICAL root: 5
2023-12-04 17:10:06,147 BMGF-R913QDCE user[11904] DEBUG user: 11
2023-12-04 17:10:06,147 BMGF-R913QDCE user[11904] INFO user: 22
2023-12-04 17:10:06,147 BMGF-R913QDCE user[11904] WARNING user: 33
2023-12-04 17:10:06,148 BMGF-R913QDCE user[11904] ERROR user: 44
2023-12-04 17:10:06,148 BMGF-R913QDCE user[11904] CRITICAL user: 55

Note: I only need CRITICAL

File: idmtools_platform_general/idmtools_platform_file/platform_operations/asset_collection_operations.py [hunk 4]
```diff
@@ -145,15 +145,19 @@ def dump_assets(self, item: Union[Experiment, Simulation], **kwargs) -> None:
             None
         """
         if isinstance(item, Experiment):
+            self.pre_create(item.assets)
```
Comment (shchen-idmod) [inline idmtools_platform_general/idmtools_platform_file/platform_operations/asset_collection_operations.py:148]: this pre_create and post_create for asset_collection seems never get used with client pre and post hook.

File: idmtools_platform_comps/idmtools_platform_comps/comps_operations/experiment_operations.py (hunk 3 above)
Comment (ZDu-IDM) [reply idmtools_platform_comps/idmtools_platform_comps/comps_operations/experiment_operations.py]: Fixed wrong parameter issue.

**Comment by ZDu-IDM** [comment]
> For log bug #2137 with test:
> 
> ```
> def test_console_off_log_off1(self):
>         ini_content = f"""
>         [Logging]
>             level = CRITICAL
>             console = on
>             enable_file_logging = off
>             user_output = on
>         """
>         self.create_temp_ini(ini_content)
>         IdmConfigParser(file_name=self.tempfile_ini.name)
>         self.platform = Platform('SLURMSTAGE')
>         logger_demo.write_some_logs(user=True, root=True, comps=True, exp=True)
> ```
> 
> Used to print: 2023-12-04 17:09:26,172 BMGF-R913QDCE root[57444] CRITICAL root: 5 2023-12-04 17:09:26,173 BMGF-R913QDCE user[57444] CRITICAL user: 55
> 
> Now print: 2023-12-04 17:10:06,146 BMGF-R913QDCE root[11904] DEBUG root: 1 2023-12-04 17:10:06,146 BMGF-R913QDCE root[11904] INFO root: 2 2023-12-04 17:10:06,146 BMGF-R913QDCE root[11904] WARNING root: 3 2023-12-04 17:10:06,147 BMGF-R913QDCE root[11904] ERROR root: 4 2023-12-04 17:10:06,147 BMGF-R913QDCE root[11904] CRITICAL root: 5 2023-12-04 17:10:06,147 BMGF-R913QDCE user[11904] DEBUG user: 11 2023-12-04 17:10:06,147 BMGF-R913QDCE user[11904] INFO user: 22 2023-12-04 17:10:06,147 BMGF-R913QDCE user[11904] WARNING user: 33 2023-12-04 17:10:06,148 BMGF-R913QDCE user[11904] ERROR user: 44 2023-12-04 17:10:06,148 BMGF-R913QDCE user[11904] CRITICAL user: 55
> 
> Note: I only need CRITICAL

Fixed.

**Review by shchen-idmod — CHANGES_REQUESTED** [review]

**Review by shchen-idmod — APPROVED** [review]

**Pull request merged**